src/lib.rs

```diff
@@ -845,6 +851,99 @@ impl JsonlToCiff {
         self
     }
 
+    /// Quantize a score to an 8-bit integer representation.
+    /// Maps the score range [min_score, max_score] to [MIN_QUANTIZED_VALUE, MAX_QUANTIZED_VALUE].
+    ///
+    /// # Arguments
+    /// * `score` - The score to quantize
+    /// * `min_score` - The minimum score in the dataset (must be > 0)
+    /// * `max_score` - The maximum score in the dataset (must be >= min_score and > 0)
+    ///
+    /// # Returns
+    /// * `0` if score <= 0 (will be filtered out)
+    /// * `MAX_QUANTIZED_VALUE` if min_score == max_score (all scores identical)
+    /// * Quantized value in range [MIN_QUANTIZED_VALUE, MAX_QUANTIZED_VALUE] otherwise
+    ///
+    /// # Panics
+    /// Panics if min_score <= 0 or max_score <= 0 or max_score < min_score
+    fn quantize_score(score: f64, min_score: f64, max_score: f64) -> i32 {
+        assert!(
+            min_score > 0.0,
+            "min_score must be greater than 0, got {}",
+            min_score
+        );
+        assert!(
+            max_score > 0.0,
+            "max_score must be greater than 0, got {}",
+            max_score
+        );
+        assert!(
+            max_score >= min_score,
+            "max_score ({}) must be >= min_score ({})",
+            max_score,
+            min_score
+        );
+
+        if score <= 0.0 {
+            0 // Will be filtered out
+        } else if min_score == max_score {
+            // If all scores are the same, map to max value
+            MAX_QUANTIZED_VALUE
```

Comment:
Just curious: would there be any difference between choosing max and min? If not, then wouldn't it be better to chose min? If yes, could you explain the difference?

Reply:
I think if `min == max ` then it does not make any difference. Sure, let's pick min as default

Reply:
Ok, thanks. Yeah, just to clarify, I suggested min because it's more space efficient in the end, though this is such an unexpected corner case, it shouldn't make any difference really...